pkg/test/framework/resource/version.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"strings"
+
+	"istio.io/istio/pilot/pkg/model"
+)
+
+// Version is an Istio version running within a cluster.
+type Version string
+
+// Versions represents a collection of Istio versions running in a cluster.
+type Versions []Version
+
+// Set parses Versions from a string flag in the form "1.5.6,1.9.0,1.4".
+func (v *Versions) Set(value string) error {
```

Comment:
You already have `ParseVersions` ... why do you need this? Are you implementing it as a flag?

Reply:
Yes, this approach parses the `IstioVersions` directly from a flag. Removed `ParseVersions` since we probably don't need it unless parsing from topoology.